### DISCUSSION TITLE
how to implement custom store?

### DISCUSSION THREAD

**Discussion by bokunodev** [tulir/whatsmeow#268] [General]
we want to use mongodb as store. since we already use mongo for many things. so why not?
but the interface is not clear imo.
```go
type IdentityStore interface {
	PutIdentity(address string, key [32]byte) error
	DeleteAllIdentities(phone string) error
	DeleteIdentity(address string) error
	IsTrustedIdentity(address string, key [32]byte) (bool, error)
}
```
there is address and phone, whats the different? whats their purpose?
also what is key?
~
what is `IdentityStore`? whats the purpose? how it should behaves?

**Comment by edwardanthony** [comment]
Any update on this?

**Comment by a5r0n** [comment]
Address is a complete JID, with agent ID, 
Phone is used to delete all user's devices identities by their phone number
